File: frontend/src/services/AcademicianService.js
import api from './api';

const AcademicianService = {
  getProfile: async () => {
    const response = await api.get('/academician/profile');
    return response.data;
  },

  updateProfile: async (profileData) => {
    const response = await api.put('/academician/profile', profileData);
    return response.data;
  },

  getMyCourses: async () => {
    const response = await api.get('/academician/courses');
    return response.data;
  },

  getCourseStudents: async (courseId) => {
    const response = await api.get(`/academician/courses/${courseId}/students`);
    return response.data;
  },

  getCourseSchedule: async (courseId) => {
    const response = await api.get(`/academician/courses/${courseId}/schedule`);
    return response.data;
  },

  getCourseExams: async (courseId) => {
    const response = await api.get(`/academician/courses/${courseId}/exams`);
    return response.data;
  },

  getAllStudents: async () => {
    const response = await api.get('/academician/all-students');
    return response.data;
  },

  getExamGrades: async (examId) => {
    const response = await api.get(`/academician/exams/${examId}/grades`);
    return response.data;
  },

  getMySchedule: async () => {
    const response = await api.get('/academician/schedule');
    return response.data;
  },

  getAdviseePerformance: async () => {
    const response = await api.get('/academician/advisees/performance');
    return response.data;
  },

  getPendingEnrollments: async () => {
    const response = await api.get('/academician/advisees/enrollments/pending');
    return response.data;
  },

  approveEnrollment: async (enrollmentId) => {
    const response = await api.put(`/academician/enrollments/${enrollmentId}/approve`);
    return response.data;
  },

  rejectEnrollment: async (enrollmentId) => {
    const response = await api.put(`/academician/enrollments/${enrollmentId}/reject`);
    return response.data;
  },

  recordAttendance: async (courseId, data) => {
    const response = await api.post(`/academician/courses/${courseId}/attendance`, data);
    return response.data;
  },

  getTodayAttendance: async (courseId) => {
    const response = await api.get(`/academician/courses/${courseId}/attendance/today`);
    return response.data;
  },

  enterGrades: async (examId, data) => {
    const response = await api.post(`/academician/exams/${examId}/grades`, data);
    return response.data;
  },

  createExam: async (courseId, data) => {
    const response = await api.post(`/academician/courses/${courseId}/exams`, data);
    return response.data;
  },

  updateExam: async (examId, data) => {
    const response = await api.put(`/academician/exams/${examId}`, data);
    return response.data;
  },

  deleteExam: async (examId) => {
    await api.delete(`/academician/exams/${examId}`);
  },

  createCourse: async (data) => {
    const response = await api.post('/academician/courses', data);
    return response.data;
  },

  postAnnouncement: async (courseId, data) => {
    const response = await api.post(`/academician/courses/${courseId}/announcements`, data);
    return response.data;
  }
};

export default AcademicianService;

File: backend/src/main/java/com/obs/backend/service/AcademicianService.java
package com.obs.backend.service;

import com.obs.backend.dto.academician.*;
import com.obs.backend.dto.user.UpdateProfileRequest;
import com.obs.backend.model.Exam;
import java.util.List;

public interface AcademicianService {
    AcademicianProfileDTO getProfile();

    void updateProfile(String username, com.obs.backend.dto.user.UpdateProfileRequest request);

    List<AcademicianCourseDTO> getMyCourses();

    List<AcademicianStudentDTO> getCourseStudents(Long courseId);

    List<AcademicianStudentDTO> getMyAdvisees();

    // Phase 1.5 - Enhanced Features
    void updateCourseSyllabus(Long courseId, CourseSyllabusRequest request);

    List<AcademicianScheduleDTO> getMySchedule();

    List<AdviseePerformanceDTO> getAdviseePerformance();

    // Phase 2
    void createExam(Long courseId, ExamRequest request);

    void updateExam(Long examId, ExamRequest request);

    void deleteExam(Long examId);

    List<AcademicianExamDTO> getCourseExams(Long courseId);

    List<GradeEntryRequest.StudentGradeDTO> getExamGrades(Long examId);

    void enterGrades(Long examId, GradeEntryRequest request);

    void recordAttendance(Long courseId, AttendanceEntryRequest request);

    List<AttendanceEntryRequest.StudentAttendanceDTO> getTodayAttendance(Long courseId);

    void openCourse(Long courseId, CourseOpenRequest request);

    void sendMessageToAdvisee(Long studentId, AdviseeMessageRequest request);

    // Phase 3 - Advanced Portal
    List<com.obs.backend.dto.academician.PendingEnrollmentDTO> getPendingAdviseeEnrollments();

    void approveEnrollment(Long enrollmentId);

    void rejectEnrollment(Long enrollmentId);

    void postAnnouncement(Long courseId, AnnouncementRequest request);

    List<com.obs.backend.model.Announcement> getCourseAnnouncements(Long courseId);

    void setOfficeHours(List<OfficeHourRequest> requests);

    List<com.obs.backend.model.OfficeHour> getMyOfficeHours();

    CourseAnalyticsResponse getCourseAnalytics(Long courseId);

    List<com.obs.backend.dto.academician.AcademicianScheduleDTO> getCourseSchedule(Long courseId);

    java.util.Map<String, List<com.obs.backend.dto.academician.AcademicianStudentDTO>> getAllStudentsGroupedByCourse();

    // Phase 4 - Final Frontier
    String exportCourseGrades(Long courseId);

    String exportAdviseeList();
}
